.github/workflows/sdk-codegen-ci.yml

```diff
@@ -0,0 +1,179 @@
+name: sdk-codegen-ci
+
+on:
+  workflow_dispatch:
+
+jobs:
+  build-aws-sdk-js-v3:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Use Node.js 14.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14.x
+
+      - name: Increase Node Heap space
+        run: echo "NODE_OPTIONS=--max-old-space-size=16384" >> $GITHUB_ENV
+
+      - uses: actions/checkout@v1
+      - id: smithy-version
+        uses: ./.github/workflows/checkout-and-build-smithy
+
+      - name: Add smithyVersion env variable
+        run: echo "smithyVersion=${{ steps.smithy-version.outputs.smithy-version }}" >> $GITHUB_ENV
+
+      - name: Checkout smithy-typescript
+        uses: actions/checkout@v2
+        with:
+          repository: awslabs/smithy-typescript
+          path: smithy-typescript
+
+      - name: Build smithy-typescript
+        run: cd smithy-typescript && ./gradlew -PsmithyVersion=$smithyVersion clean build publishToMavenLocal
+
+      - name: Checkout aws-sdk-js-v3
+        uses: actions/checkout@v2
+        with:
+          repository: aws/aws-sdk-js-v3
+          path: aws-sdk-js-v3
+
+      - name: Build aws-sdk-js-v3
+        run: cd aws-sdk-js-v3/codegen && ./gradlew -PsmithyVersion=$smithyVersion clean build
+
+      - name: Test aws-sdk-js-v3
+        run: cd aws-sdk-js-v3 && yarn && yarn test:all
+
+  build-aws-sdk-go-v2:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-java@v2
+        with:
+          java-version: 17
+          distribution: 'zulu'
+
+      - name: Set up Go 1.19
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.19
+
+      - uses: actions/checkout@v1
+      - id: smithy-version
+        uses: ./.github/workflows/checkout-and-build-smithy
+
+      - name: Add smithyVersion env variable
+        run: echo "smithyVersion=${{ steps.smithy-version.outputs.smithy-version }}" >> $GITHUB_ENV
+
+      - name: Checkout smithy-go
+        uses: actions/checkout@v2
+        with:
+          repository: aws/smithy-go
+          path: smithy-go
+
+      - name: Build smithy-go
+        run: cd smithy-go/codegen && ./gradlew -PsmithyVersion=$smithyVersion clean build publishToMavenLocal
+
+      - name: Checkout aws-sdk-go-v2
+        uses: actions/checkout@v2
+        with:
+          repository: aws/aws-sdk-go-v2
+          path: aws-sdk-go-v2
+
+      - name: Build aws-sdk-go-v2
+        run: cd aws-sdk-go-v2/codegen && ./gradlew -PsmithyVersion=$smithyVersion clean build -Plog-tests && ./gradlew clean
+
+      - name: Run aws-sdk-go-v2 protocol tests
+        # This is essentially the `ci-test` command in the Makefile
```

Comment:
Then why aren't we just running that?

Reply:
Because [ci-test](https://github.com/aws/aws-sdk-go-v2/blob/main/Makefile#L340) uses [generate](https://github.com/aws/aws-sdk-go-v2/blob/main/Makefile#L80) which uses [smithy-generate](https://github.com/aws/aws-sdk-go-v2/blob/main/Makefile#L84). 

The smithy version needs to be overriden via `./gradlew -PsmithyVersion=$smithyVersion clean build`,  the `smithy-generate` command doesn't allow that and hence I had to split up the commands from `ci-test`.